manifests/kustomize/base/installs/multi-user/pipelines-profile-controller/sync.py

```diff
@@ -98,7 +101,141 @@ def server_factory(visualization_server_image,
+            json.dump(mc_config, outfile)
+        admin = MinioAdmin(target="kubeflow-minio", binary_path='/app/mc', config_dir='/tmp')
+
+        def upsert_lifecycle_policy(self, bucket_name):
+            lfc = {
+                "Rules": [
+                    {
+                        "Status": "Enabled",
+                        "Filter": {"Prefix": "private-artifacts/"},
+                        "Expiration": {"Days":31},
+                        "ID": "private-artifacts",
+                    },
+                ]
+            }
+            print('upsert_lifecycle_policy:', lfc)
+            api_response = self.s3_client.put_bucket_lifecycle_configuration(Bucket=bucket_name,
+                                                                             LifecycleConfiguration=lfc)
+            print(api_response)
+
+        def upsert_iam_policy(self, shared_bucket_name):
+            namespace_isolation_policy = {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "s3:GetBucketLocation",
+                        ],
+                        "Resource": [
+                            "arn:aws:s3:::%s"  % shared_bucket_name
+                        ]
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "s3:ListBucket",
+                        ],
+                        "Resource": [
+                            "arn:aws:s3:::%s"  % shared_bucket_name # the root is needed to list the bucket
+                        ],
+                        "Condition":{"StringLike":{"s3:prefix": [
+                            "", "shared/*", "artifacts/*","private-artifacts/", "private/",
+                            "private-artifacts/${aws:username}/*", "private/${aws:username}/*"
+                        ]}}
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "s3:ListBucket",
+                            "s3:GetBucketLocation",
+                            "s3:GetBucketPolicy",
+                            "s3:GetObject",
+                            "s3:DeleteObject",
+                            "s3:PutObject"
+                        ],
+                        "Resource": [
+                            "arn:aws:s3:::%s/artifacts/*"  % shared_bucket_name, # old shared artifacts for backwards compatibility
+                            "arn:aws:s3:::%s/private-artifacts/${aws:username}/*"  % shared_bucket_name, # private artifacts
+                            "arn:aws:s3:::%s/private/${aws:username}/*"  % shared_bucket_name, # private storage
+                            "arn:aws:s3:::%s/shared/*"  % shared_bucket_name # shared storage for collaboration
```

Comment:
This is AWS specific usage, cc @chensun to comment on this.

Reply:
I would like to double click on this comment in case we missed it. Although this overlay is marked as `minio` overlay, I noticed that there are some platform specific logic as highlighted here: `arn:aws:s3` is an AWS S3 resource. Please make sure this overlay works in minio environment.

Reply:
Yes it does work perfectly. Minio is explicitly designed to be compatible with that. Myself and other users are using that PR for several months in production. But I had a long discussion with @surajkota From The AWS Kubeflow distribution and their requirements. They are also unsatisfied with the current state. There are three different architectures regarding storage and we would like to streamline that for "the same user journey". This also covers successors for the AGPL minio and minio gateway. I will present this overhaul in detail here soon.